contracts/programs/gumball-machine/src/lib.rs

```diff
@@ -426,6 +446,42 @@ pub mod gumball_machine {
         bubblegum::cpi::create_tree(cpi_ctx, max_depth, max_buffer_size)
     }
 
+    /// Initialize chunk of NFT indices (as many as possible within the compute budget of a single transaction). All indices must be initialized before the tree can dispense.
+    pub fn initialize_indices_chunk(
+        ctx: Context<InitIndices>
+    ) -> Result<()> {
+        // Fetch mutable header data
+        let mut gumball_machine_data = ctx.accounts.gumball_machine.try_borrow_mut_data()?;
+        let (mut header_bytes, config_data) =
+            gumball_machine_data.split_at_mut(std::mem::size_of::<GumballMachineHeader>());
+        let mut gumball_header = GumballMachineHeader::load_mut_bytes(&mut header_bytes)?;
+
+        // Assert that indices initialization is authorized
+        assert_eq!(gumball_header.authority, ctx.accounts.authority.key());
+
+        // Grab mutable reference to indices bytes
+        let size = gumball_header.max_items as usize;
+        let index_array_size = std::mem::size_of::<u32>() * size;
+        let (indices_data, _) = config_data.split_at_mut(index_array_size);
+        let indices = cast_slice_mut::<u8, u32>(indices_data);
+
+        // Determine the next byte range to initialize
+        let first_index_to_initialize = gumball_header.smallest_uninitialized_index as usize;
+        let next_smallest_uninitialized_index = (gumball_header.smallest_uninitialized_index + 250000).min(gumball_header.max_items) as usize;
```

Comment:
not a huge fan of the hardcoded 250000

Reply:
i'll pull this into `state` as a constant